cmd/controller-manager/app/controller_manager.go

```diff
@@ -293,58 +293,76 @@ func StartControllers(s *options.ControllerManagerServer,
 	recorder record.EventRecorder,
 	stop <-chan struct{}) error {
 
-	// Get available service-catalog resources
-	glog.V(5).Info("Getting available resources")
-	availableResources, err := getAvailableResources(serviceCatalogClientBuilder)
+	// When Catalog Controller and Catalog API Server are started at the
+	// same time with API Aggregation enabled, it may take some time before
+	// Catalog registration shows up in API Server.  Attempt to get resources
+	// every 10 seconds and quit after 3 minutes if unsuccessful.
+
+	backoff := wait.Backoff{
+		Duration: 10 * time.Second,
+		Factor:   1,
+		Jitter:   0,
+		Steps:    19,
+	}
+	var availableResources map[schema.GroupVersionResource]bool
+	err := wait.ExponentialBackoff(
```

Comment:
`wait.PollImmediate` more closely matches what we are doing here. Since the factor of the exponential backoff is 1, it looks deceiving to use `wait.ExponentialBackoff`.